src/functional/ffront/type_info.py

```diff
@@ -286,6 +307,35 @@ def is_concretizable(symbol_type: ct.SymbolType, to_type: ct.SymbolType) -> bool
     return False
 
 
+def _is_zero_dim_field(field: ct.SymbolType) -> bool:
+    return isinstance(field, ct.FieldType) and field.dims != Ellipsis and len(field.dims) == 0
+
+
+def is_zero_dim_field_broadcastable(a_arg: ct.SymbolType, b_arg: ct.SymbolType) -> bool:
+    """
+    Check if first argument is a zero-dimensional field and second is of ScalarType.
+
+    If arguments have the same dtype, the latter is broadcastable to a zero-dimensional field.
+    """
+    if is_number(b_arg) and extract_dtype(a_arg) == extract_dtype(b_arg):
+        return True
+    elif is_number(b_arg):
+        raise GTTypeError(f"{b_arg} is not compatible with {a_arg}")
+    return False
+
+
+def promote_zero_dims(
+    args: list[ct.SymbolType], function_type: ct.FieldOperatorType | ct.ProgramType
+):
+    """Cast arg types to zero dimensional fields if compatible and required by function signature."""
+    new_args = args
```

Comment:
This is not sufficient to avoid modifying the argument of the caller. See this example:
```
>>> a = [1, 2, 3]
>>> b = a
>>> b[0] = 5
>>> a[0]
5  # -> a has only changed
```
You can either do a shallow copy or what I would suggest is this pattern:
```python
new_args = []
for arg in args:
    if ...:
        new_arg = ...
        new_args.append(new_arg)
    else:
        new_args.append(arg)